commons/src/main/java/io/aiven/kafka/connect/common/config/SinkCommonConfig.java

```diff
@@ -0,0 +1,197 @@
+/*
+ * Copyright 2024 Aiven Oy
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.aiven.kafka.connect.common.config;
+
+import java.time.ZoneId;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.stream.Collectors;
+
+import org.apache.kafka.common.config.ConfigDef;
+import org.apache.kafka.common.config.ConfigException;
+
+import io.aiven.kafka.connect.common.config.validators.FileCompressionTypeValidator;
+import io.aiven.kafka.connect.common.config.validators.OutputFieldsEncodingValidator;
+import io.aiven.kafka.connect.common.config.validators.OutputFieldsValidator;
+import io.aiven.kafka.connect.common.config.validators.OutputTypeValidator;
+import io.aiven.kafka.connect.common.grouper.RecordGrouperFactory;
+import io.aiven.kafka.connect.common.templating.Template;
+
+import edu.umd.cs.findbugs.annotations.SuppressFBWarnings;
+
+public abstract class SinkCommonConfig extends CommonConfig {
+    public static final String FORMAT_OUTPUT_FIELDS_CONFIG = "format.output.fields";
```

Comment:
There are several fields like this FORMAT_OUTPUT_FIELDS_CONFIG and methods like addOutputFieldsFormatConfigGroup, which are s3 specific. While we are creating this new s3-commons module now, why not move also all these s3 specific things or this SinkCommonConfig moved to s3-commons ?

Reply:
I should have mentioned this before, sorry.

Reply:
Hey Murali these are actually used across the sink connectors, for both gcs and azure.